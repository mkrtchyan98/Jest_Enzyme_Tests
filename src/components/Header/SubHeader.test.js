import React from 'react';
import { shallow } from 'enzyme';
import SubHeader,{ Title,SubHeaderButton } from './SubHeader';


describe('<SubHeader />', () => {
	it('should render', () => {
		const component = shallow(<SubHeader />);

		expect(component).toMatchSnapshot();
	});

	it('should render with a dynamic title', () => {
		const title = 'Test Complate';
		const component = shallow(<SubHeader title={title}  />);

		expect(component.find(Title).text()).toEqual(title);
	});

	it('should render with  buttons and handle the onClick events',() => {
		const mockGoBack = jest.fn();
	    const mockOpenForm = jest.fn();

		const component = shallow(<SubHeader openForm={mockOpenForm}   goBack={mockGoBack}/>);

		const goBackButton = component.find(SubHeaderButton);
		const goBackButton = component.find(SubHeaderButton).at(0);
		expect(goBackButton.exists()).toBe(true);

		const mockOpenForm = component.find(SubHeaderButton);
		const mockOpenForm = component.find(SubHeaderButton).at(1);
		expect(mockOpenForm.exists()).toBe(true);

		goBackButton.simulate('click');
		expect(mockGoBack).toHaveBeenCalled();

		mockOpenForm.simulate('click');
		expect(mockOpenForm).toHaveBeenCalled();

	});
 
	})
