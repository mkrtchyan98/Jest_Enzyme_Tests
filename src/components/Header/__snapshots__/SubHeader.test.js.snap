// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubHeader /> should render 1`] = `ShallowWrapper {}`;

exports[`<SubHeader /> should render with a dynamic title 1`] = `ShallowWrapper {}`;

exports[`<SubHeader /> should render with a form button 1`] = `ShallowWrapper {}`;

exports[`<SubHeader /> should render with a goBack button 1`] = `ShallowWrapper {}`;
